src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -331,33 +331,30 @@ private function getDriver(): string
             $this->driver = $con->getAttribute(\PDO::ATTR_DRIVER_NAME);
         } else {
             $driver = $con->getDriver();
+            $platform = $driver->getDatabasePlatform();
+
+            if ($driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver) {
+                throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
+            }
 
             switch (true) {
-                case $driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver:
-                    throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
-                case $driver instanceof \Doctrine\DBAL\Driver\AbstractMySQLDriver:
+                case $platform instanceof \Doctrine\DBAL\Platforms\MySQLPlatform:
                     $this->driver = 'mysql';
                     break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOSqlite\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\SQLite\Driver:
+                case $platform instanceof \Doctrine\DBAL\Platforms\SqlitePlatform:
                     $this->driver = 'sqlite';
                     break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOPgSql\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\PgSQL\Driver:
+                case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQLPlatform:
```

Comment:
```suggestion
                case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQLPlatform:
                case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQL94Platform:
```

For DBAL 3.1 😓

Reply:
what ? They removed the base unversioned class ? that's quite bad newws

Reply:
Reverted back in 3.2, and deprecated the whole "platform classes for specific versions": https://github.com/doctrine/dbal/commit/666c0656c03587802b1524a9b54e8a104a6f03b1